homeassistant/components/plugwise/binary_sensor.py

```diff
@@ -96,3 +112,52 @@ def _async_process_data(self):
             self._icon = FLAME_ICON if self._is_on else IDLE_ICON
 
         self.async_write_ha_state()
+
+
+class PwNotifySensor(PwBinarySensor, BinarySensorEntity):
```

Comment:
im wondering if this entity should be opt-in (ie disabled by default). maybe this should be a device action/event so people can handle this how they desire. @MartinHjelmare have you ever dealt with a decision?

Reply:
i just feel this is more of an open proxy for notifications compared to how such are normally used

Reply:
We shouldn't be generating notifications from the entity itself.   If we want to generate notifications they shouldn't be coming from the entity.

Reply:
Do you guys mean to state that Notifications are meant for only showing Core-related messages? And not for showing messages that are copied from an entity?

Reply:
@ctalkington disabled by default is a good idea.

@bdraco We are doing that: we are checking for a message being present via `async_update_data`. If there is one found then the message is relayed as a Persistent Notification.

A little more background: Plugwise Notifications can easily be missed as most of them are like debug-type messages. Showing them as a Persistent Notification in Core makes it possible to actually notice these messages and have a look at their content. Maybe not everyone wants this so the disabled by default suggestion is a good idea.

Reply:
Why would we show debug-type messages as persistent notifications? That doesn't make any sense for normal users.

Reply:
I wouldn't classify them debug messages, but to give an example of what @bouwew means (from [fixture](https://github.com/home-assistant/core/blob/b79dc911463467c937d023f4aaa9b7df1137855e/tests/fixtures/plugwise/adam_multiple_devices_per_zone/notifications.json)) 
`"warning": "Node Plug (with MAC address 000D6F000D13CB01, in room 'n.a.') has been unreachable since 23:03 2020-01-18. Please check the connection and restart the device."`

Reply:
Tom, thanks for correcting me: the messages are indeed warning messages, using the word "debug" was not correct.

Reply:
So that's not something the integration should warn about. The entities should be marked as unavailable instead. It would be up to Home Assistant to provide a way to deal with this as the user wants, that spans all devices/entities.

Reply:
@balloob that is true for that particular warning. But there can be other warnings as well like "the (water) pressure in your heating system is too low". Or, "the heating system reports an error", "the pump has not been running for 24hrs, running it for 2 mins to avoid rust-forming". Some messages are more serious of nature than others.

Reply:
Before I reply, any other context that I am missing or wrong word used somewhere 😉

Reply:
The scope has been clarified enough I think. Awaiting you final judgement ;)

Reply:
I don't think there is more to add - just summarizing - there was a finding that these 'notifications' don't follow through within HA and Bouwe took the opportunity to build this so it would be clear that there were important problems. So the 'missing plug' fixture might be a less great example but the 'reporting error' or 'pump' things definitely are. But we'll leave it up to Core if this is the right path for carrying such notices to the user. So indeed, awaiting judgement 😅

Reply:
In my opinion these cases should be represented as binary sensors and/or timestamp sensor with latest reported data if that's not already available in the state machine. Then the user can create automations based off that.

Reply:
With all of the above, 5fbd885 removes said functionality leaving in the `binary_sensor`

Reply:
Please let us know if there needs to be more time to validate/discuss this or removing the persistent notifications (as per last commits) is acceptable. We can set this PR to draft (or close and recall it later) so other Plugwise PRs we have ready to draft and queue can go first?